app/models/draw_suites_update.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+#
+# Form / service object to handle the updating of draw suites
+class DrawSuitesUpdate
+  include ActiveModel::Model
+
+  attr_reader :size, :suite_ids, :drawn_suite_ids, :undrawn_suite_ids
+
+  # permit :update to be called on the parent class
+  def self.update(**params)
+    new(**params).update
+  end
+
+  # Initialize a new DrawSuitesUpdate
+  #
+  # @param draw [Draw] the draw to be updated
+  # @param params [#to_h] the parameters from the form
+  def initialize(draw:, params: nil)
+    @draw = draw
+    @suite_ids = draw.suite_ids
+    process_params(params) if params
+  end
+
+  # Execute the suites update, remove all suites to be removed and add all
+  # suites to be added. Occurs in a transaction for safety.
+  #
+  # @return [Hash{Symbol=>String,Hash,Nil,DrawStringUpdate] a result hash
+  #   containing the appropriate path to redirect to, a flash message to set,
+  #   and the DrawSuitesUpdate object if there were any failures.
+  def update
+    return no_action_warning if suites_to_remove.empty? && suites_to_add.empty?
+    ActiveRecord::Base.transaction do
+      remove_suites unless suites_to_remove.empty?
+      add_suites unless suites_to_add.empty?
+    end
+    success
+  rescue ActiveRecord::RecordInvalid => error
+    error(error)
+  end
+
+  private
+
+  attr_reader :draw, :params, :suites_to_add, :suites_to_remove
+
+  def process_params(params)
+    @params = params.to_h.transform_keys(&:to_sym)
+    @size = extract_size
+    update_ids_param(:suite_ids)
+    update_ids_param(:drawn_suite_ids)
+    update_ids_param(:undrawn_suite_ids)
+    @suites_to_remove = find_suites_to_remove
+    @suites_to_add = find_suites_to_add
+  end
+
+  def extract_size
+    raise ArgumentError if size_param_not_an_int
```

Comment:
I mean, if we don't raise here I'm pretty sure that the queries that make use of the size will raise errors anyway since ActiveRecord won't be able to cast the param to an integer. We can talk more about this after pilot.